src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2808,6 +2808,10 @@ protected function castAttribute($key, $value)
                 return json_decode($value, true);
             case 'collection':
                 return new BaseCollection(json_decode($value, true));
+            case 'date':
+                return Carbon::createFromFormat('Y-m-d', $value);
+            case 'datetime':
+                return Carbon::createFromFormat('Y-m-d H:i:s', $value);
```

Comment:
You're saying that grammar can provide the format to pass into Carbon::createFromFormat()? or because the grammar is different it should just use the value and let Carbon handle it?

If its the first one can you provide a link to where grammar does that?